packages/govuk-frontend/src/govuk/settings/_typography-responsive.scss

```diff
@@ -151,4 +277,30 @@ $govuk-typography-scale: (
       line-height: 1.2
     )
   )
-) !default;
+);
+
+/// Responsive typography font map
+///
+/// This is used to generate responsive typography that adapts according to the
+/// breakpoints.
+///
+/// Font size and font weight can be defined for each breakpoint. You can define
+/// different behaviour on tablet and desktop. The 'null' breakpoint is for
+/// mobile.
+///
+/// Line-heights will automatically be converted from pixel measurements into
+/// relative values. For example, with a font-size of 16px and a line-height of
+/// 24px, the line-height will be converted to 1.5 before output.
+///
+/// You can also specify a separate font size and line height for print media.
+///
+/// @type Map
+///
+/// @prop {Number} $point.$breakpoint.font-size - Font size for `$point` at `$breakpoint`
+/// @prop {Number} $point.$breakpoint.line-height - Line height for `$point` at `$breakpoint`
+/// @prop {Number} $point.print.font-size - Font size for `$point` when printing
+/// @prop {Number} $point.print.line-height - Line height for `$point` when printing
+///
+/// @access public
+
+$govuk-typography-scale: if($govuk-new-typography-scale, $_new-typography-scale, $_old-typography-scale) !default;
```

Comment:
This PR looks pretty ace

Wondered if we should tweak these variable names to follow the colour palette switch?

Bit like with CSS, adding the `-legacy` or `-modern` modifiers at the end, but `$_govuk-` at the start

```scss
$govuk-colours: if(
  $govuk-use-legacy-palette,
  $_govuk-colour-palette-legacy,
  $_govuk-colour-palette-modern
) !default;
```

You can see when we removed it in https://github.com/alphagov/govuk-frontend/commit/2a0a53908682d2b04f1e3660f63e766f67bd6515